nixos/modules/services/misc/netbox-agent.nix

```diff
@@ -0,0 +1,80 @@
+{ config, pkgs, lib, ... }:
+let
+  cfg = config.services.netbox-agent;
+  settingsFormat = pkgs.formats.json {};
```

Comment:
There is a `pkgs.formats.yaml`, as you say in the settings option that it will be converted to yaml